domain/src/main/java/com/teamagam/gimelgimel/domain/base/interactors/BaseDisplayInteractor.java

```diff
@@ -0,0 +1,27 @@
+package com.teamagam.gimelgimel.domain.base.interactors;
+
+import com.teamagam.gimelgimel.domain.base.executor.PostExecutionThread;
+import com.teamagam.gimelgimel.domain.base.executor.ThreadExecutor;
+
+public abstract class BaseDisplayInteractor extends BaseInteractor {
+
+    private final ThreadExecutor mThreadExecutor;
+    private final PostExecutionThread mPostExecutionThread;
+
+    public BaseDisplayInteractor(
+            ThreadExecutor threadExecutor,
+            PostExecutionThread postExecutionThread) {
+        mThreadExecutor = threadExecutor;
+        mPostExecutionThread = postExecutionThread;
+    }
+
+    @Override
+    protected final Iterable<SubscriptionRequest> buildSubscriptionRequests() {
+        return buildSubscriptionRequests(
+                new DisplaySubscriptionRequest.DisplaySubscriptionRequestFactory(mThreadExecutor,
+                        mPostExecutionThread));
+    }
+
+    protected abstract Iterable<SubscriptionRequest> buildSubscriptionRequests(
```

Comment:
I'm not sure about this Iterable thing. Every interactor should return Collection.toSingleton?